snippets/volcano_analysis.R

```diff
@@ -0,0 +1,58 @@
+library(tidyverse)
+library(lubridate)
+library(ggplot2)
+library(maps)
+library(patchwork)
+
+#' This is script with analysis of volcanos for my
+#' geological article.
+
+volcano <- readr::read_csv('https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2020/2020-05-12/volcano.csv')
+
+processed_volcano <- volcano %>%
+  mutate(logpop10 = log10(population_within_10_km+0.1)) %>%
+  select(primary_volcano_type, pop=logpop10, longitude, latitude)
+
+# cleaning data a bit as there are some plural versions of volcano names
+# that mean essentially the same
+processed_volcano$primary_volcano_type <-
+  recode(processed_volcano$primary_volcano_type, Stratovolcano="Stratovolcano(es)")
+
+processed_volcano$primary_volcano_type <-
+  recode(processed_volcano$primary_volcano_type, `Shield(s)`="Shield")
+
+volcano_types <- processed_volcano %>% group_by(primary_volcano_type) %>%
+  summarise(count = n()) %>%
+  arrange(desc(count)) %>% head(8)
+
+# NA types become Other as we want to include them in the analysis
+processed_volcano$primary_volcano_type <- processed_volcano$primary_volcano_type %>%
+  replace_na("Other")
+
+sc_max <- max(processed_volcano$pop)
+sc_min <- min(processed_volcano$pop)
+
+col_palette = c("#d11141", "#00b159", "#00aedb", "#f37735", "#ffc425",
```

Comment:
Perhaps better to use an existing colour palette. Indexing col_palette in this case doesn't work if colid > 15.